qubit-sdk/src/main/java/com/qubit/android/sdk/internal/placement/interactor/PlacementInteractorImpl.kt

```diff
@@ -0,0 +1,67 @@
+package com.qubit.android.sdk.internal.placement.interactor
+
+import com.qubit.android.sdk.api.placement.PlacementMode
+import com.qubit.android.sdk.api.placement.PlacementPreviewOptions
+import com.qubit.android.sdk.api.tracker.OnPlacementError
+import com.qubit.android.sdk.api.tracker.OnPlacementSuccess
+import com.qubit.android.sdk.internal.configuration.repository.ConfigurationModel
+import com.qubit.android.sdk.internal.configuration.repository.ConfigurationRepository
+import com.qubit.android.sdk.internal.placement.PlacementImpl
+import com.qubit.android.sdk.internal.placement.callback.PlacementCallbackConnectorImpl
+import com.qubit.android.sdk.internal.placement.connector.PlacementConnectorBuilder
+import com.qubit.android.sdk.internal.placement.model.PlacementModel
+
+internal class PlacementInteractorImpl(
+    placementConnectorBuilder: PlacementConnectorBuilder,
+    private val configurationRepository: ConfigurationRepository,
+    private val deviceId: String
+) : PlacementInteractor {
+
+  companion object {
+    private val DEFAULT_PLACEMENT_MODE = PlacementMode.LIVE
+  }
+
+  private val placementConnector = placementConnectorBuilder.build()
+
+  override fun fetchPlacement(
+      placementId: String,
+      mode: PlacementMode?,
+      previewOptions: PlacementPreviewOptions,
+      onSuccess: OnPlacementSuccess,
+      onError: OnPlacementError
+  ) {
+    placementConnector.getPlacementModel(
+        getPlacementApiHost(configurationRepository),
```

Comment:
`fetchPlacement` is method called by the user of SDK in any thread he want including main thread. It should not have any I/O operations. But here, we have reading from SharedPreferences, which is I/O operation (potentially, if not cached).
Do you think is it ok?

In EventTracker I did it totally asynchronously. And there was assumption, not to read from repository, from module which isn't 'owner' of it. All interactions with configurations is done via ConfigurationService.
There was also an assumption (important in case of events. I don't know how about placements), that if SDK didn't fetch configuration from yet, functions of SDK can be used, but execution are be postponed.